yt_dlp/postprocessor/embedthumbnail.py

```diff
@@ -224,4 +224,8 @@ def run(self, info):
             thumbnail_filename if converted or not self._already_have_thumbnail else None,
             original_thumbnail if converted and not self._already_have_thumbnail else None,
             info=info)
+
+        if not self._already_have_thumbnail:
+            info['thumbnails'][idx].pop('filepath', None)
```

Comment:
I'm using the `.pop(name, None)` pattern a few places to prevent raising an error if a key doesn't exist. While it shouldn't be strictly necessary based on my testing, there is _so much_ I don't know about yt-dlp that I figured this was the safer bet